src/proc_self_mountinfo.h

```diff
@@ -1,55 +1,106 @@
 #ifndef NETDATA_PROC_SELF_MOUNTINFO_H
 #define NETDATA_PROC_SELF_MOUNTINFO_H 1
+/**
+ * @file proc_self_mountinfo.h
+ * @brief This file holds the API, used to find mounted filesystems.
+ *
+ * mountinfo_read() reads the file /proc/mountinfo and stores it's content in a linked list.
+ * Entries of this list can be searched with mountinfo_find*().
+ *
+ * To free the list, run mountinfo_free().
+ */
 
-#define MOUNTINFO_IS_DUMMY      0x00000001
-#define MOUNTINFO_IS_REMOTE     0x00000002
-#define MOUNTINFO_IS_BIND       0x00000004
-#define MOUNTINFO_IS_SAME_DEV   0x00000008
-#define MOUNTINFO_NO_STAT       0x00000010
-#define MOUNTINFO_NO_SIZE       0x00000020
-#define MOUNTINFO_READONLY      0x00000040
+#define MOUNTINFO_IS_DUMMY      0x00000001 ///< dummy mountinfo
+#define MOUNTINFO_IS_REMOTE     0x00000002 ///< mounts a remote device
+#define MOUNTINFO_IS_BIND       0x00000004 ///< mountinfo is bind
+#define MOUNTINFO_IS_SAME_DEV   0x00000008 ///< mount is on same dev
+#define MOUNTINFO_NO_STAT       0x00000010 ///< cannot `stat()` mountinfo
+#define MOUNTINFO_NO_SIZE       0x00000020 ///< size not available
+#define MOUNTINFO_READONLY      0x00000040 ///< mounted readonly
 
+/** One mountpoint */
 struct mountinfo {
-    long id;                // mount ID: unique identifier of the mount (may be reused after umount(2)).
-    long parentid;          // parent ID: ID of parent mount (or of self for the top of the mount tree).
-    unsigned long major;    // major:minor: value of st_dev for files on filesystem (see stat(2)).
-    unsigned long minor;
+    long id;                ///< mount ID: unique identifier of the mount (may be reused after umount(2)).
+    long parentid;          ///< parent ID: ID of parent mount (or of self for the top of the mount tree).
+    unsigned long major;    ///< major: value of st_dev for files on filesystem @see man 2 stat
+    unsigned long minor;    ///< minor: value of st_dev for files on filesystem @see man 2 stat
 
-    char *persistent_id;    // a calculated persistent id for the mount point
-    uint32_t persistent_id_hash;
+    char *persistent_id;         ///< a calculated persistent id for the mount point
+    uint32_t persistent_id_hash; ///<hash of `president_id`
 
-    char *root;             // root: root of the mount within the filesystem.
-    uint32_t root_hash;
+    char *root;             ///< root of the mount within the filesystem.
+    uint32_t root_hash;     ///< hash of `root`
 
-    char *mount_point;      // mount point: mount point relative to the process's root.
-    uint32_t mount_point_hash;
+    char *mount_point;         ///< mount point relative to the process's root.
+    uint32_t mount_point_hash; ///< hash of `mount_point`
 
-    char *mount_options;    // mount options: per-mount options.
+    char *mount_options; ///< per-mount options.
 
-    int optional_fields_count;
+    int optional_fields_count; ///< count optional fields \todo deprecated?
```

Comment:
`/proc/mountinfo` may have a number of optional fields, that are terminated with a field with contents ` - `. This counts how many of such fields are there. I have never found a system that this is non-zero.